### PR TITLE
Add mechanism to set data boundaries.

### PR DESCRIPTION
**Pull request by danh32**
Closes: https://github.com/robinhoodmarkets/spark/issues/2

Let me know what you think about this one. I caved and added a `Bounds` type to avoid a whole mess of trouble in the unit tests with mocking (since RectF is in android.jar). I'm not sure it's a good idea to introduce our own type, but it does let us be more direct with the value names (`minX` and `minY` instead of `left` and `right`).

### CODE REVIEW

File: spark/src/main/java/com/robinhood/spark/SparkAdapter.java [hunk 1]
```diff
@@ -18,22 +18,22 @@
...
- * data is changed.
+ * A simple adapter class - evenly distributes your points along the x axis, does not draw a base
+ * line, and has support for registering/notifying {@link DataSetObserver}s when data is changed.
  */
 public abstract class SparkAdapter {
     private final DataSetObservable observable = new DataSetObservable();
 
     /**
-     * returns the number of points to be drawn
+     * Gets the number of points to be drawn
```
Comment (jinatonic) [inline spark/src/main/java/com/robinhood/spark/SparkAdapter.java]: nit @return

File: spark/src/main/java/com/robinhood/spark/SparkAdapter.java [hunk 2]
```diff
@@ -18,22 +18,22 @@
...
      */
     public abstract int getCount();
 
     /**
-     * returns the object at the given index
+     * Gets the object at the given index
```
Comment (jinatonic) [inline spark/src/main/java/com/robinhood/spark/SparkAdapter.java]: here too

File: spark/src/main/java/com/robinhood/spark/SparkAdapter.java [hunk 3]
```diff
@@ -50,7 +50,36 @@ public float getX(int index) {
...
 
     /**
-     * Return true if you wish to draw a "baseLine" - a horizontal line across the graph used
+     * Gets the float representation of the boundaries of the entire dataset. By default, this will
+     * be the min and max of the actual data points in the adapter. This can be overridden for
+     * custom behavior.
+     *
+     * @return the bounds desired around this adapter's data
+     */
+    public Bounds getDataBounds() {
```
Comment (jinatonic) [inline spark/src/main/java/com/robinhood/spark/SparkAdapter.java]: can this be `protected`?

Comment (jinatonic) [inline spark/src/main/java/com/robinhood/spark/SparkAdapter.java]: seems silly to use a `Bounds` class, why not a `Rect`?

File: spark/src/main/java/com/robinhood/spark/Bounds.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package com.robinhood.spark;
+
+/**
+ * Value class that holds the min and max x and y values for the data set to be graphed.
+ */
+public final class Bounds {
```
Comment (jinatonic) [inline spark/src/main/java/com/robinhood/spark/Bounds.java]: Seems super unnecessary, why not just use a `RectF`?

File: spark/src/test/java/com/robinhood/spark/SparkAdapterUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package com.robinhood.spark;
+
+import org.junit.Test;
+
+import static junit.framework.Assert.assertEquals;
+
+public class SparkAdapterUnitTest {
+
+    @Test
+    public void testGetDataBounds() {
```
Comment (jinatonic) [inline spark/src/test/java/com/robinhood/spark/SparkAdapterUnitTest.java]: add edge-case test cases, e.g. no data and negative data.

File: spark/src/main/java/com/robinhood/spark/SparkAdapter.java (hunk 3 above)
Comment (danh32) [inline spark/src/main/java/com/robinhood/spark/SparkAdapter.java]: I think to get the behavior that the requester wanted, he'll need to get access to this bounds from a handful of adapters, to determine the union of all the adapters and then set that bounds on every adapter. Would definitely make things easier to leave this public.

File: spark/src/test/java/com/robinhood/spark/SparkAdapterUnitTest.java (hunk 5 above)
Comment (danh32) [inline spark/src/test/java/com/robinhood/spark/SparkAdapterUnitTest.java]: Good call! the negative data test caught a bug.

**Comment by danh32** [comment]
New approach without the `Bounds` class, per our offline discussion.
